packages/trpc/server/routers/publicViewer/noShow.handler.ts

```diff
@@ -31,33 +45,66 @@ export const noShowHandler = async ({ input }: NoShowOptions) => {
           email: true,
         },
       });
-
+      const allAttendeesMap = allAttendees.reduce((acc, attendee) => {
+        acc[attendee.email] = attendee;
+        return acc;
+      }, {} as Record<string, { id: number; email: string }>);
       const updatePromises = attendees.map((attendee) => {
-        const attendeeToUpdate = allAttendees.find((a) => a.email === attendee.email);
-
-        if (attendeeToUpdate) {
-          return prisma.attendee.update({
-            where: { id: attendeeToUpdate.id },
-            data: { noShow: attendee.noShow },
-          });
-        }
+        const attendeeToUpdate = allAttendeesMap[attendee.email];
+        if (!attendeeToUpdate) return;
+        return prisma.attendee.update({
+          where: { id: attendeeToUpdate.id },
+          data: { noShow: attendee.noShow },
+        });
       });
-
-      await Promise.all(updatePromises);
-    } else {
-      await prisma.booking.update({
-        where: {
-          uid: bookingUid,
-        },
-        data: {
-          noShowHost: true,
+      const results = await Promise.allSettled(updatePromises);
+      const _attendees = results
+        .filter((x) => x.status === "fulfilled")
+        .map((x) => (x as PromiseFulfilledResult<{ noShow: boolean; email: string }>).value)
+        .map((x) => ({ email: x.email, noShow: x.noShow }));
+      const payload = await getResultPayload(_attendees);
+      const booking = await prisma.booking.findUnique({
+        where: { uid: bookingUid },
+        select: {
+          eventType: {
+            select: {
+              id: true,
+              teamId: true,
+              userId: true,
+            },
+          },
         },
       });
+      const orgId = await getOrgIdFromMemberOrTeamId({
+        memberId: booking?.eventType?.userId,
+        teamId: booking?.eventType?.teamId,
+      });
```

Comment:
These are required payloads for Webhooks service. Would like to simplify even more.